pkg/scheduler/framework/runtime/framework.go

```diff
@@ -730,11 +730,10 @@ func (f *frameworkImpl) RunFilterPlugins(
 			if !pluginStatus.IsUnschedulable() {
 				// Filter plugins are not supposed to return any status other than
 				// Success or Unschedulable.
-				errStatus := framework.AsStatus(fmt.Errorf("running %q filter plugin: %w", pl.Name(), pluginStatus.AsError())).WithFailedPlugin(pl.Name())
-				return map[string]*framework.Status{pl.Name(): errStatus}
+				pluginStatus = framework.AsStatus(fmt.Errorf("running %q filter plugin: %w", pl.Name(), pluginStatus.AsError()))
 			}
 			pluginStatus.SetFailedPlugin(pl.Name())
-			statuses[pl.Name()] = pluginStatus
+			return map[string]*framework.Status{pl.Name(): pluginStatus}
```

Comment:
We don't need the `statuses` at all, just return nil at the end.

As a followup, we should remove the map completely, but lets leave it for now to make it easy to backport.

Reply:
> We don't need the statuses at all, just return nil at the end.

To fix the issue _only_, I'd like to leave all cleanup work in a follow-up. If we return nil, UTs need to be tweaked, but what concerned me more is the downstream users - like the users using v1.24 and suppose it's always returning a concrete map.

Reply:
ok, if that breaks the tests, then we can do that in a followup.

Reply:
FYI: changing the return type will also require to tweak codes in cluster-autoscaler.